src/engine/strat_engine/engine.rs

```diff
@@ -65,25 +73,41 @@ impl StratEngine {
             return Err(EngineError::Engine(ErrorEnum::Error, err_msg));
         }
 
-        let pools = find_all()?;
-
-        let mut table = Table::default();
-        for (pool_uuid, devices) in &pools {
-            let evicted = table.insert(StratPool::setup(*pool_uuid, devices)?);
-            if !evicted.is_empty() {
-
-                // TODO: update state machine on failure.
-                let _ = teardown_pools(table.empty());
+        let mut engine = StratEngine {
```

Comment:
I don't think it's necessary to make the StratEngine struct until after you've processed the pools. The reason for this is that I don't think you would handle an event until after the pool is done initializing.

Reply:
This comment is not entirely clear to me, but I've moved the creation to where it was before, I've put it closer to the way it was before.

It's entirely possible that when we start the daemon and do a device scan that we only find one device of a multi-device pool, thus we will be placing that device into the map of incomplete pools, this has nothing to do with event handing.  However, this discussion does remind me that the order is very important and that we just overlooked a race condition when bringing up the daemon.  We must register for udev events before we walk the device tree, otherwise a device could show up just after the scan but before we register for events, thus causing us to miss a device and fail to bring up a pool.  I will reorder stratisd.rs to account for this.